File: SparseArray/LL.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <cstring>
using namespace std;

struct NodeString{
        int frequency;
        char * name;
        NodeString * next;
                
        NodeString(int _frequency, char * _name)
            {
            frequency = _frequency;
            name = _name;
            }
};

class UnrollList{
    private:
        NodeString * startNode;
    public:
        UnrollList(){};
        ~UnrollList(){};

        void addStart(NodeString &newStart){
            if(startNode){
                startNode -> next = &newStart;
            }
            else{
                startNode = &newStart;
            }
        }


};

class LL{
    private:
        //UnrollList* internalList[];
        std::vector<UnrollList> internalList;
    public:
        LL(){
            //Constructor
            //internalList[20] = new UnrollList();
            
            internalList = vector<UnrollList>(20);
        }
        ~LL(){};
        void add(char* string){
            /*
            //get the length
            int position = strlen(string);
            
            //use the length to find the array 
            //if the size already exists, then
            if(internalList(position)){
                
                //check if this string already exists
                NodeString * temp;
                temp = internalList[position] -> startNode;
                while(temp -> next != NULL){
                    if(temp-> name == string){
                        temp -> frequency += 1;
                        break;
                    }
                    temp -> next = temp -> next -> next;
                }
                //if you haven't found it
                NodeString newString(1, string);
                temp -> next = &newString;
            }
            else{
                NodeString newString(1, string);
                internalList[position] -> addStart(newString);
            }
            */
        }   
};

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */   
    LL* sparseArray = new LL();
    //sparseArray.add("test");
    
    delete(sparseArray);
    return 0;
}

File: SparseArray/LL.py
class NodeString:
	
	def __init__(self, _frequency, _name):
		self.frequency = _frequency
		self.name = _name
		self.next

class UnrollList:

	def __init__(self):
		self.startNode

	def addStart(self, _newNode):
		if(self.startNode):
			self.startNode.next = _newNode
		else:
			self.startNode = _newNode

class LL:
	def __init__(self):
		self.internalList = [None] * 20

	#Find the string's length, and use the length
	#to find new node
	def add(self, string):
		length = len(string)

		if self.internalList[length]:
			ULL = self.internalList[length].startNode
			if ULL:
				if ULL.startNode.name == string:
					ULL.frequency += 1
				else:
					newString = NodeString(1, string)
					ULL.next = newString
		else:
			newString = NodeString(1,string)
			self.internalList[length] = newString



if __name__ == "__main__":
	print "Hello, Testing"
	sparseArray = LL();

	sparseArray.add("abc")
